python-client/giskard/push/prediction.py

```diff
@@ -33,26 +34,22 @@ def create_overconfidence_push(model, ds, df):
 
 
 def create_borderline_push(model, ds, df):
-    if model.meta.model_type == SupportedModelTypes.CLASSIFICATION:
-        # row_slice = ds.slice(lambda x: x.loc[[idrow]], row_level=False)
-        row_slice = Dataset(df=df, target=ds.target, column_types=ds.column_types.copy(), validation=False)
-        model_prediction_results = model.predict(row_slice)
-        all_predictions = model_prediction_results.all_predictions
-        diff, max, second = _var(all_predictions)
-        values = row_slice.df
-        training_label = values[ds.target].values[0]
-        training_label_proba = model_prediction_results.all_predictions[training_label].values[0]
-
-        if diff <= 0.1:
-            return BorderlinePush(max, second, training_label, training_label_proba, row_slice)
-
-
-def _var(x):
-    row_as_list = x.values.flatten().tolist()
-    max_val = max(row_as_list)
-    row_as_list.remove(max_val)
-    second_max_val = max(row_as_list)
-    diff = max_val - second_max_val
-    # diff = abs(max_val - second_max_val)/second_max_val
-    # diff = abs(max_val - second_max_val)
-    return diff, max_val, second_max_val
+    row_slice = Dataset(df=df, target=ds.target, column_types=ds.column_types.copy(), validation=False)
+    prediction_results = model.predict(row_slice)
+    values = row_slice.df
+    target_value = values[ds.target].values.item()
+
+    if model.is_classification:
+        target_value_proba = prediction_results.all_predictions[target_value].values.item()
+        if len(model.meta.classification_labels) > 2 or model.meta.classification_threshold is None:
+            sorted_predictions = np.sort(prediction_results.raw[0])
+            abs_diff = sorted_predictions[-1] - sorted_predictions[-2]
```

Comment:
This should be `abs_diff = sorted_predictions[-2] - sorted_predictions[-1]` since they're sorted otherwise it's negative.

what if `len(model.meta.classification_labels) == 3`, why not also check if `abs_diff = sorted_predictions[-3] - sorted_predictions[-2]` <= 0.1?

what if `len(model.meta.classification_labels) == N`, why not also check if `abs_diff = sorted_predictions[-N] - sorted_predictions[-N-1]` <= 0.1 for all Ns?

Reply:
I believe that 
```python
abs_diff = sorted_predictions[-2] - sorted_predictions[-1]
```  
is positive as the last value is higher than the penultimate one

Reply:
About the other N -1 cases, the goal is to identify the examples that can switch from one prediction to another easily, I'm not sure that we want to use the N-1 other features

Reply:
> I believe that
> 
> ```python
> abs_diff = sorted_predictions[-2] - sorted_predictions[-1]
> ```
> 
> is positive as the last value is higher than the penultimate one

Ah you're right, the sorting is descending. Sorry.



> About the other N -1 cases, the goal is to identify the examples that can switch from one prediction to another easily, I'm not sure that we want to use the N-1 other features

Okay you're right.